hi-ml-azure/src/health_azure/himl.py

```diff
@@ -704,8 +705,17 @@ def create_v2_inputs(ml_client: MLClient, input_datasets: List[DatasetConfig]) -
     inputs: Dict[str, Input] = {}
     for i, input_dataset in enumerate(input_datasets):
         input_name = f"INPUT_{i}"
-        version = input_dataset.version or 1
-        data_asset: Data = ml_client.data.get(input_dataset.name, version=str(version))
+        version = input_dataset.version
+        # if version is None, this function gets the latest version
+        logging.info(
+            f"Trying to access data asset {input_dataset.name} version {version}, datastore {input_dataset.datastore}"
+        )
+        data_asset: Data = _get_or_create_v2_data_asset(
+            ml_client,
+            input_dataset.datastore,
+            input_dataset.name,
+            version=str(version),
```

Comment:
if version is None this will be casted as 'None', so the check line 176 won't be valid, is that intended?
<img width="344" alt="image" src="https://user-images.githubusercontent.com/37396332/215083766-94522fe6-9fe5-47dd-bb56-71d7efb7a1be.png">

Reply:
Nope not intended, just me being lazy when trying to fix pyright. Good spot, thank you!